lib/active_model/serializer/adapter/json_sideload.rb

```diff
@@ -0,0 +1,206 @@
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonSideload < Json
+
+        # the list of what has already been serialized will be kept here to
+        # help avoid infinite recursion
+        #
+        # this should be a list of association_name to a list of objects
+        attr_accessor :serialized
+
+        # When we are sideloading associations, we can more easily track what
+        # has been serialized, so that we avoid infinite
+        # recursion / serialization.
+        def initialize(*args)
+          @serialized = {}
+
+          super(*args)
+        end
+
+        def serializable_hash(options = nil)
+          options ||= {}
+
+          # begin recursive serialization
+          # TODO: this will evaluate ALL specified relationships
+          # throughout the entire tree.
+          #
+          # Do we want a way to limit this?
+          serialize_hash(options)
```

Comment:
what if, for a particular serializer's associations, depth could be defined by using the `include` keyword?

something like this:

```ruby
class BlogSerializer < ...
  has_many :posts, include: [:author, comments: :authors]
```

that way, we'd only get exactly what we expect in the json?

 - 1 blog
 - however many authors
 - however many comments